code/game/objects/structures/door_assembly.dm

```diff
@@ -165,143 +170,188 @@
 
 /obj/structure/door_assembly/attackby(obj/item/W as obj, mob/user as mob)
 	if(W.ispen())
-		var/t = sanitizeSafe(input(user, "Enter the name for the door.", src.name, src.created_name), MAX_NAME_LEN)
-		if(!t)	return
-		if(!in_range(src, usr) && src.loc != usr)	return
-		created_name = t
-		return
+		var/door_name = sanitizeSafe(input(user, "Enter the name for the door.", src.name, src.created_name), MAX_NAME_LEN)
+		if(!door_name)
+			return
+		if(!Adjacent(user))
+			return
+		created_name = door_name
 
-	if(W.iswelder() && ( (istext(glass)) || (glass == 1) || (!anchored) ))
+	else if(W.iswelder())
+		if(!(istext(glass) || glass == TRUE || !anchored))
+			to_chat(user, SPAN_WARNING("\The [src] isn't ready to be welded yet. It has to have glass installed to remove the glass, or it has to be unsecured to deconstruct."))
```

Comment:
This is still a very cryptic message. It has to have glass installed _to remove_ the glass?